### PR TITLE
Update libsbp interfaces.

### PR DESCRIPTION
**Pull request by gsmcmullin**
Updates to match swift-nav/libsbp#228.

/cc @mfine @mookerji

### CODE REVIEW

**Comment by mfine** [comment]
This is looking great! Nice cleanup!

File: piksi_tools/action_logger.py [hunk 1]
```diff
@@ -254,9 +254,9 @@ def main():
           link.add_callback(sl.printer, SBP_MSG_PRINT_DEP)
```
Comment (mookerji) [inline piksi_tools/action_logger.py]: Do we want to apply the Forwarder to the other callbacks (printer, etc.)?

File: piksi_tools/bootload.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def handshake(self, timeout=None):
       t0 = time.time()
     self.handshake_received = False
     expire = time.time() + 15.0
-    self.link.send(SBP_MSG_RESET, "")
+    self.link(MsgReset())
```
Comment (mookerji) [inline piksi_tools/bootload.py]: nit: Actually, I'm pretty sure there's a `self.link.reset()` that does just this.

File: piksi_tools/bootload.py [hunk 3]
```diff
@@ -92,13 +92,13 @@ def handshake(self, timeout=None):
         self.link.send(SBP_MSG_RESET, "")
```
Comment (mookerji) [inline piksi_tools/bootload.py]: ditto on reset

**Comment by mookerji** [comment]
This is phenomenal.

File: piksi_tools/action_logger.py (hunk 1 above)
Comment (gsmcmullin) [inline piksi_tools/action_logger.py]: We can and should apply the Forwarder wherever the callbacks can block or take time to run.  There is no real gain where they are quick, or just post something to run on the UI thread.

File: piksi_tools/bootload.py (hunk 2 above)
Comment (gsmcmullin) [inline piksi_tools/bootload.py]: Hmm.  There was, but it disappeared in my work on the iterators and isn't in 0.50.1.  This will have to do for now.

**Pull request merged**